ProditionsService/inc/ProditionsHandle.hh

```diff
@@ -19,10 +20,20 @@ namespace mu2e {
     typedef std::shared_ptr<ENTITY> ptr_t;
     typedef std::shared_ptr<const ENTITY> cptr_t;
 
-    ProditionsHandle() {
-      ENTITY e;
+    ProditionsHandle():ptr(nullptr) {
+      // find the name of the ENTITY and strip the namespace
+      _name =  boost::typeindex::type_id<ENTITY>().pretty_name();
+      if(_name.find("::")!=std::string::npos) _name=_name.erase(0,_name.find("::")+2);
```

Comment:
boost claims that it works "with many compilers" and will work with gcc, except if you force RTTI off, and then only in special cases.  If you force RTTI off, then dynamic_cast won't work, so it seems unlikely we would do that.  In any case, if the mechanism fails, it will not fail silently.  But if you don't like this solution, we can talk about other ones.

Reply:
I once used std::type_info and typeid to print the name of a template argument; as I recall it was a bit ugly when the type itself was templated but it basically worked.   Is this use case something similar?  an example is at:
https://en.cppreference.com/w/cpp/language/typeid

Reply:
The key part of Ray's answer is that it's not likely to fail silently. There is no sense in putting effort into understanding robustness unless it's likely to fail silently.  So that part of this issue is resolved.  Dave's comment is ongoing.